flink-streaming-connectors/flink-connector-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/HBaseSink.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.hbase;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
+import org.apache.flink.util.Preconditions;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.client.Durability;
+import org.apache.hadoop.hbase.client.Increment;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+public class HBaseSink<IN> extends RichSinkFunction<IN> {
+	private static final long serialVersionUID = 1L;
+
+	private static final Logger LOG = LoggerFactory.getLogger(HBaseSink.class);
+
+	private transient HBaseClient client;
+	private String tableName;
+	private HBaseMapper<IN> mapper;
+	private boolean writeToWAL = true;
+
+	public HBaseSink(String tableName, HBaseMapper<IN> mapper) {
+		Preconditions.checkArgument(!Strings.isNullOrEmpty(tableName), "Table name cannot be null or empty");
+		Preconditions.checkArgument(mapper != null, "HBase mapper cannot be null");
+		this.tableName = tableName;
+		this.mapper = mapper;
+	}
+
+	public HBaseSink writeToWAL(boolean writeToWAL) {
+		this.writeToWAL = writeToWAL;
+		return this;
+	}
+
+	@Override
+	public void open(Configuration configuration) throws Exception {
+		try {
+			// use config files found in the classpath
+			client = new HBaseClient(HBaseConfiguration.create(), tableName);
+		} catch (IOException e) {
+			throw new RuntimeException("HBase sink preparation failed.", e);
+		}
+	}
+
+	@Override
+	public void invoke(IN value) throws Exception {
+		byte[] rowKey = mapper.rowKey(value);
+		List<Mutation> mutations = Lists.newArrayList();
+		for (HBaseMapper.HBaseColumn column : mapper.columns(value)) {
+			Mutation mutation;
+			if (column.isStandard()) {
```

Comment:
Better to created Enums for PUT, DELETE, INCREMENT and APPEND and handle all of them here? so that Table#batch() will work for all. But Deletes are much more complicated because there are ROW deletes, COL deletes , family deletes, deletes with specific versions.